exonum/src/api/websockets.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! WebSocket API.
+
+use actix::*;
+use actix_web::ws;
+
+use rand::{self, Rng, ThreadRng};
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+use api::ServiceApiState;
+use crypto::Hash;
+
+/// WebSocket Message for communication between Clients(`WsSession`) and Server(`WsServer`).
+#[derive(Message, Debug)]
+pub(crate) struct Message(pub String);
+
+#[derive(Message)]
+#[rtype(usize)]
+pub(crate) struct Subscribe {
+    pub address: Recipient<Syn, Message>,
+}
+
+#[derive(Message)]
+pub(crate) struct Unsubscribe {
+    pub id: usize,
+}
+
+#[derive(Message)]
+pub(crate) struct Broadcast {
+    pub block_hash: Hash,
+}
+
+/// WebSocket server.
+pub(crate) struct WsServer {
+    pub subscribers: HashMap<usize, Recipient<Syn, Message>>,
+    rng: RefCell<ThreadRng>,
+}
+
+impl Default for WsServer {
+    fn default() -> Self {
+        Self {
+            subscribers: HashMap::new(),
+            rng: RefCell::new(rand::thread_rng()),
+        }
+    }
+}
+
+impl Actor for WsServer {
+    type Context = Context<Self>;
+}
+
+impl Handler<Subscribe> for WsServer {
+    type Result = usize;
+
+    fn handle(&mut self, msg: Subscribe, _ctx: &mut Self::Context) -> usize {
+        let id = self.rng.borrow_mut().gen::<usize>();
+        self.subscribers.insert(id, msg.address);
+
+        id
+    }
+}
+
+impl Handler<Unsubscribe> for WsServer {
+    type Result = ();
+
+    fn handle(&mut self, msg: Unsubscribe, _ctx: &mut Self::Context) {
+        let Unsubscribe { id } = msg;
```

Comment:
You can move this let's inside handle signature. 
something like this should work`fn handle(&mut self, Unsubscribe { id }: Unsubscribe, _ctx: &mut Self::Context) {`